django/cantusdb_project/main_app/templates/chant_detail.html

```diff
@@ -444,17 +448,25 @@ <h4>List of melodies</h4>
                                     <ul>
                                         {% for editor in source.proofreaders.all %}
                                             <li>
-                                                <a href="{% url 'user-detail' editor.id %}"><b>{{ editor.full_name }}</b></a>
-                                                <br>
+                                                {% if editor.full_name %}
+                                                    <a href="{% url 'user-detail' editor.id %}"><b>{{ editor.full_name }}</b></a>
+                                                {% else %}
+                                                    <a href="{% url 'user-detail' editor.id %}"><b>User {{ editor.id }}</b></a>
+                                                {% endif %}
                                             </li>
+                                            <br>
                                         {% endfor %}
                                     </ul>
                                 {% endif %}
 
                                 {{ source.indexing_notes|default_if_none:"" }}
                                 <br>
                                 {% if source.created_by %}
-                                    Contributor: <a href={% url 'user-detail' source.created_by.id %}><b>{{ source.created_by.full_name }}</b></a>
+                                    {% if source.created_by.full_name %}
```

Comment:
Just a flag to remind you to make sure this model traversal (`a.b.c`) is optimized. This sort of code is exactly the kind that will introduce lots of SQL queries if you're not careful.